### DISCUSSION TITLE
distinguish various twinning systems

### DISCUSSION THREAD

**Discussion by KKone121** [mtex-toolbox/mtex#2274] [Ask Anything] [answered]
Hi!
I have introduced different twins such as {10-12},{10-11} in Mg  alloys by compression.
I want to **distinguish different kinds of twins** in different colors and calculate the twin area fraction.
Help is appreciated, thank you.

**Comment by kilir** [comment]
Hi,
can you distinguish which part of the grain is the twin and which part is the host?
Cheers,
Rüdiger

**Reply by KKone121** [reply]
Hi, Rüdiger.
By referring to the examples of mtex, I can distinguish the twins and the host. It's hard that I can't distinguish between different twin types.
Kyle

**Reply by kilir** [reply]
Hi,
while I don't recall there's a section elaborating that, in any case, if you know the different twin laws, simply discriminate the boundaries accordingly e.g.

```Matlab
twinBoundary1 = gB(angle(gB.misorientation,twinlaw1) < 5*degree);
twinBoundary2 = gB(angle(gB.misorientation,twinlaw2) < 5*degree);
```
If you have some prior knowledge what is the host and what is the twin, you can then apply it to the neighboring grains of your twin boundaries. You'll get the unique grain ids from `unique(twinBoundary.grainId,'row')`, discreminite twin and host and compute the volume fraction from those.

Cheers,
Rüdiger

**Reply by KKone121** [reply]
Thanks for your kind suggestions, and I'm trouble in how to definition the twinning law of 11-21 twin now.
As described in [](https://mtex-toolbox.github.io/orientation.map.html), I have coded how to define 11-21 twin, but it always reports errors. The code is as follows, could u give me some suggestions?
```Matlab
twinning1 = orientation.map(Miller(0,1,-1,-2,CS),Miller(0,-1,1,-2,CS),Miller(2,-1,-1,0,CS),Miller(2,-1,-1,0,CS));
twinning2 = orientation.map(Miller(1,1,-2,1,CS),Miller(1,-1,2,-1,CS), Miller(1,-1,0,0,CS),Miller(-1,1,0,0,CS));
```
_The error is_ Inconsitent pairs of vectors! The angle between u1, u2 and v1, v2 needs to be the same, but differs by 61.9451°. I think it may be a crystallographic knowledge, but without any ideas

**Reply by kilir** [reply, marked as answer]
How about 
`twinning2 = orientation.map(Miller(1,1,-2,1,CS),Miller(-1,-1,2,-1,CS), Miller(1,-1,0,0,CS),Miller(-1,1,0,0,CS));` ?

**Reply by KKone121** [reply]
That's a good idea, thanks for u, Rüdiger.

I would like to ask how I can calculate the twin area by using misorientation to distinguish different twins. In the [Mtex merging grains section](https://mtex-toolbox.github.io/GrainMerge.html), it mentions using twins that occupy less than half of the total area for identification, but this does not seem to align with my situation.

Additionally, I have not found similar cases in Mtex regarding the use of the unique function 'unique(twinBoundary.grainId,'row')'. Could you please provide a more detailed explanation?

Thank you for your guidance.

Kyle

**Reply by KKone121** [reply]
Here is the code to supplement my original idea：
`twin1fraction=sum(area(grains(isTwinning1)))/sum(area(grains)) * 100;
twin2fraction=sum(area(grains(isTwinning2)))/sum(area(grains)) * 100;`

**Comment by apedgaon** [comment]
@KKone121 @kilir, how do you construct the misorientation object for a twin? How do you know it is (11-21) twin and then how do you infer the right misorientation for it? I am also observing the twinning2 in my EBSD data but unsure which type of twin it is. The misorientation object seems to transform the twin plane (11-21) which doesn't seem logical.
